test/cql-pytest/test_prepare.py

```diff
@@ -150,3 +150,46 @@ def test_composite_pk_indexes_duplicate_named_variables(cql, table2, scylla_only
     prepared = cql.prepare(
         f"SELECT * FROM {table2} WHERE p1 = :x AND p2 = :x AND p3 = :z AND p4 = :y AND c1 = :y AND c2 = :z ")
     assert prepared.routing_key_indexes == [0, 0, 1, 2]
+
+# Test what happens when using a bind marker with the same name (e.g., ":x")
+# twice in a query. Above we tested which "routing_key_indexes" is returned
+# in a PREPARE request, but here we test the functionality of the duplicate
+# marker when actually executing the query.
+# Note: The CQL protocol allows using bind markers in both QUERY (unprepared
+# statement) and EXECUTE (prepared statement) cases, but it appears that the
+# Python driver only supports bind markers with prepared statements (and
+# for unprepared statements uses a different mechanisms with Python "%s" and
+# "%(name)s"), so this test is only for the prepared-statement case.
+# Reproduces issue #15559.
+@pytest.mark.xfail(reason="Different behavior in Scylla and Cassandra - see issue #15559")
+def test_duplicate_named_bind_marker_prepared(cql, table1):
+    x = unique_key_int()
+    cql.execute(f'INSERT INTO {table1} (p,c) VALUES ({x},{x})')
+    cql.execute(f'INSERT INTO {table1} (p,c) VALUES ({x},{x+1})')
+    # Sanity check: query without bind markers, with unnamed bind markers,
+    # and with two different bind-marker names. All should work.
+    assert [(1,1)] == list(cql.execute(f'SELECT * FROM {table1} WHERE p={x} AND c={x}'))
```

Comment:
I think there's a typo here, it should be `(x, x)` instead of `(1, 1)`. AFAIU `unique_key_int` can return a different value when other tests are ran in parallel.

Reply:
Right. Thanks for noticing. I missed this problem because I ran this test alone (so x=1), then when I noticed it was failing I marked it xfail, and then when run after (not even in parallel!) with other tests, it fail even more badly but then I didn't know.